File: Front/proyecto-software-frontend/app/services/authService.js
import apiClient from "../api/axios";
import { API_PREFIX } from "./config";

const BASE = `${API_PREFIX}/auth`;

export const authService = {
  register(payload) {
    return apiClient.post(`${BASE}/register`, payload);
  },

  verifyEmail(token) {
    return apiClient.get(`${BASE}/verify-email`, { params: { token } });
  },

  login(credentials) {
    return apiClient.post(`${BASE}/login`, credentials);
  },

  refresh(refreshToken) {
    return apiClient.post(`${BASE}/refresh`, { refreshToken });
  },

  logout(refreshToken) {
    const body = refreshToken ? { refreshToken } : undefined;
    return apiClient.post(`${BASE}/logout`, body);
  },

  forgotPassword(correo) {
    return apiClient.post(`${BASE}/forgot-password`, { correo });
  },

  resetPassword(payload) {
    return apiClient.post(`${BASE}/reset-password`, payload);
  },

  me() {
    return apiClient.get(`${BASE}/me`);
  },
};

export default authService;

File: backend/src/main/java/com/proyectoarquitectura/app/service/auth/AuthService.java
package com.proyectoarquitectura.app.service.auth;

import com.proyectoarquitectura.app.models.dto.auth.AuthResponse;
import com.proyectoarquitectura.app.models.dto.auth.LoginRequest;
import com.proyectoarquitectura.app.models.dto.auth.RegisterRequest;
import com.proyectoarquitectura.app.models.dto.auth.UsuarioResponse;
import jakarta.servlet.http.HttpServletRequest;

public interface AuthService {

    UsuarioResponse register(RegisterRequest req);

    void verifyEmail(String token);

    AuthResponse login(LoginRequest req, HttpServletRequest httpRequest);

    AuthResponse refresh(String refreshToken);

    void logout(String refreshToken);

    void forgotPassword(String correo);

    void resetPassword(String token, String nuevaContrasena);
}
